packages/widget-reorder/src/widget.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect } from 'react'
+import { fromJS } from 'immutable'
+import { DropResult } from 'react-beautiful-dnd'
+// @ts-ignore
+import { WidgetProps } from '@ncwidgets/common-typings'
+import { reducer } from './reducer';
+import { queryData, setOrder, handleDragEnd } from './action'
+import { renderDefaultPreview, PreviewContainer, getPreview } from './preview'
+import { ControlList, ControlDraggableItem, renderDefaultControl } from './control'
+
+const initialState = { 
+  data: {},
+  order: [],
+  orderModified: false,
+}
+
+export interface CreateControlOptions {
+  renderControl?: (item: Record<string, any>) => React.ReactElement
+  renderPreview?: (items: object[]) => React.ReactElement
+  name?: string
+}
+
+type CreateWidget = (options: CreateControlOptions) => React.StatelessComponent<WidgetProps>
+
+export const createWidget = ({ 
+  renderControl = renderDefaultControl,
+  renderPreview = renderDefaultPreview,
+  name = 'ncw-reorder'
+}) => {
+
+  const previewRef = React.createRef<HTMLDivElement>()
+
+  const Control  = React.forwardRef((props: WidgetProps, ref) => {
```

Comment:
Haha accidentally left it there after some testing, however it removes one of the two warnings in the console to forward a ref there. No idea why since the ref isn't even used, can it be something with the `dnd`-package?

>Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?